### PR TITLE
add resource owner password flow example req and resp

### PR DESCRIPTION
**Pull request by mystiberry-okta**
## Description:
Support requested a resource owner password flow example request and response.

## Type of Pull Request:

- [x] Content (Documentation updates or typo-fixes)
- [ ] Blog Post
- [ ] Functional (CSS changes, JS updates, or layout modifications)

### Resolves:
* [OKTA-124310](https://oktainc.atlassian.net/browse/OKTA-12431)

## How Has This Been Tested?
- [x] I have built this locally and verified that it does not break existing functionality.
- [ ] For functional changes, I have tested against Chrome, Firefox, and Safari, and mobile.
- [ ] For functional changes, I have added tests.

## Primary Reviewers:
@johngronberg-okta

### CODE REVIEW

**Comment by mystiberry-okta** [comment]
@johngronberg-okta  I had branching issues so redid this work. Please review.

File: _source/_docs/api/resources/oauth2.md [hunk 1]
```diff
@@ -532,6 +532,47 @@ invalid_grant           | The *code* or *refresh_token* value was invalid, or th
...
 
+#### Request Example: Resource Owner Password Flow
+
+~~~
+curl -X POST \
+  https://danger.oktapreview.com/oauth2/aus8tf1sb1BJ7vywW0h7/v1/token \
+  --header "accept: application/json" \
+  --header "cache-control: no-cache" \
+  --header "content-type: application/x-www-form-urlencoded" \
+  -d 'grant_type=password&username=<username>&password=<password>&scope=api%3Aread&client_id=<clientid>&client_secret=<secret>'
```
Comment (johngronberg-okta) [inline _source/_docs/api/resources/oauth2.md]: the request scopes should match the response scopes (the response scope is openid)

**Comment by johngronberg-okta** [comment]
@mystiberry-okta check out this branch: https://github.com/okta/okta.github.io/compare/ROPF_example?expand=1

**Comment by mystiberry-okta** [comment]
@johngronberg-okta I need an approval before rc will merge this.

**Comment by johngronberg-okta** [comment]
LGTM, though make sure to squash commits.

**Comment by rchild-okta** [comment]
This also needs to be rebased off of weekly, and the PR needs to point to the weekly branch

**Comment by mystiberry-okta** [comment]
I forgot to edit the base branch before trying rebase. Will fix.

**Pull request merged**